### ISSUE TITLE
app-portage/hackport-0.5.5-r1::gentoo fails to compile with ghc 8.4.3

### ISSUE THREAD

**Issue by wmyrda** [gentoo-haskell/gentoo-haskell#826]
Hence I am on stable portage-2.3.40-r1 I keep using hackport-0.5.5. Gentoo version differs from this overlay only in dependencies. Nothing related to the compilation process. Nevertheless compilation fails at the end. For some reason it keeps going regardless of the errors in the middle of compilation process.

```
diff -Naur /var/lib/layman/haskell/app-portage/hackport/hackport-0.5.5.ebuild /mnt/4tb/portage/app-portage/hackport/hackport-0.5.5-r1.ebuild 
--- /var/lib/layman/haskell/app-portage/hackport/hackport-0.5.5.ebuild	2018-04-22 07:46:30.502521298 +0200
+++ /mnt/4tb/portage/app-portage/hackport/hackport-0.5.5-r1.ebuild	2018-03-21 19:38:54.000000000 +0100
@@ -19,7 +19,12 @@
 
 RESTRICT=test # tests are broken: need path to ebuild tree
 
-RDEPEND="|| ( <sys-apps/portage-2.3.0 app-portage/repoman )
+RDEPEND="
+	|| (
+		<sys-apps/portage-2.3.0
+		sys-apps/portage-mgorny
+		app-portage/repoman
+	)
 	dev-haskell/async:=
 	dev-haskell/base16-bytestring:=
 	dev-haskell/base64-bytestring:=
```

```
[ 61 of 294] Compiling Distribution.Parsec.Class ( cabal/Cabal/Distribution/Parsec/Class.hs, dist/build/hackport/hackport-tmp/Distribution/Parsec/Class.dyn_o )
[ 62 of 294] Compiling Distribution.Text ( cabal/Cabal/Distribution/Text.hs, dist/build/hackport/hackport-tmp/Distribution/Text.dyn_o )
[ 63 of 294] Compiling Portage.Use      ( Portage/Use.hs, dist/build/hackport/hackport-tmp/Portage/Use.dyn_o )

Portage/Use.hs:39:43: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Use.hs:1:8-18
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Use.hs:13:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
39 | showModificator (X u)     = Disp.char '!' <> DT.disp u
   |                                           ^^

Portage/Use.hs:40:39: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Use.hs:1:8-18
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Use.hs:13:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
40 | showModificator (Q u)     = DT.disp u <> Disp.char '?'
   |                                       ^^

Portage/Use.hs:41:39: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Use.hs:1:8-18
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Use.hs:13:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
41 | showModificator (E u)     = DT.disp u <> Disp.char '='
   |                                       ^^

Portage/Use.hs:42:43: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Use.hs:1:8-18
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Use.hs:13:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
42 | showModificator (N u)     = Disp.char '-' <> DT.disp u
   |                                           ^^
[ 64 of 294] Compiling Language.Haskell.Extension ( cabal/Cabal/Language/Haskell/Extension.hs, dist/build/hackport/hackport-tmp/Language/Haskell/Extension.dyn_o )
[ 65 of 294] Compiling Distribution.Types.Version ( cabal/Cabal/Distribution/Types/Version.hs, dist/build/hackport/hackport-tmp/Distribution/Types/Version.dyn_o )
[ 66 of 294] Compiling Distribution.Types.VersionRange ( cabal/Cabal/Distribution/Types/VersionRange.hs, dist/build/hackport/hackport-tmp/Distribution/Types/VersionRange.dyn_o )
[ 67 of 294] Compiling Distribution.Types.VersionInterval ( cabal/Cabal/Distribution/Types/VersionInterval.hs, dist/build/hackport/hackport-tmp/Distribution/Types/VersionInterval.dyn_o )
[ 68 of 294] Compiling Distribution.Version ( cabal/Cabal/Distribution/Version.hs, dist/build/hackport/hackport-tmp/Distribution/Version.dyn_o )
[ 69 of 294] Compiling Portage.Version  ( Portage/Version.hs, dist/build/hackport/hackport-tmp/Portage/Version.dyn_o )

Portage/Version.hs:61:17: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
61 |     dispVer ver <> dispC c <> dispSuf suf <> dispRev rev
   |                 ^^

Portage/Version.hs:61:28: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
61 |     dispVer ver <> dispC c <> dispSuf suf <> dispRev rev
   |                            ^^

Portage/Version.hs:61:43: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
61 |     dispVer ver <> dispC c <> dispSuf suf <> dispRev rev
   |                                           ^^

Portage/Version.hs:67:34: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
67 |       dispRev n = Disp.text "-r" <> Disp.int n
   |                                  ^^

Portage/Version.hs:78:35: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
78 |     Alpha n -> Disp.text "_alpha" <> dispPos n
   |                                   ^^

Portage/Version.hs:79:35: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
79 |     Beta n  -> Disp.text "_beta"  <> dispPos n
   |                                   ^^

Portage/Version.hs:80:35: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
80 |     Pre n   -> Disp.text "_pre"   <> dispPos n
   |                                   ^^

Portage/Version.hs:81:35: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
81 |     RC n    -> Disp.text "_rc"    <> dispPos n
   |                                   ^^

Portage/Version.hs:82:35: error:
    Ambiguous occurrence ‘<>’
    It could refer to either ‘Prelude.<>’,
                             imported from ‘Prelude’ at Portage/Version.hs:12:8-22
                             (and originally defined in ‘GHC.Base’)
                          or ‘Text.PrettyPrint.<>’,
                             imported from ‘Text.PrettyPrint’ at Portage/Version.hs:26:26-29
                             (and originally defined in ‘Text.PrettyPrint.HughesPJ’)
   |
82 |     P  n    -> Disp.text "_p"     <> dispPos n
   |                                   ^^
[ 70 of 294] Compiling Distribution.Types.TestType ( cabal/Cabal/Distribution/Types/TestType.hs, dist/build/hackport/hackport-tmp/Distribution/Types/TestType.dyn_o )
[ 71 of 294] Compiling Distribution.Types.SourceRepo ( cabal/Cabal/Distribution/Types/SourceRepo.hs, dist/build/hackport/hackport-tmp/Distribution/Types/SourceRepo.dyn_o )
[ 72 of 294] Compiling Distribution.Types.PkgconfigName ( cabal/Cabal/Distribution/Types/PkgconfigName.hs, dist/build/hackport/hackport-tmp/Distribution/Types/PkgconfigName.dyn_o )
```

PS I am not sure how hackport-0.5.6 would behave as I would like to avoid going to unstable portage as there is still bugs open related to sync-verification, so fixing compilation for hackport-0.5.5 would be welcomed

**Comment by trofi** [comment]
It's not related to unstable `portage`. No need to use unstable portage. `hackport-0.5.5` and `hackport-0.5.6` are both ~arch thus `0.5.6` should be fine. The breakage is caused by ghc-8.4. I'll fix hackport to build against `ghc-8.4` and cut `0.5.7` version.

**Comment by trofi** [comment]
Pushed `hackport-0.5.6` to `::gentoo`: https://gitweb.gentoo.org/repo/gentoo.git/commit/app-portage/hackport/hackport-0.5.6.ebuild?id=989094927eb9d90cac212c037c893172e310a11a
